### ISSUE TITLE
Linking an action to a cluster

### ISSUE THREAD

**Issue by ashbowe** [nmrf/sadata#30]
Hi Timo,

When creating a new action I link it to a thematic cluster and I thought this would automatically link it to all of the recommendations within that cluster but this does not appear to be the case. Should it be doing this? 

Thanks

**Comment by ashbowe** [comment]
Or is it just done via the recommendations drop down where the user can select a thematic cluster and tick all or some of the recommendations within that? If so, if all are selected is it possible that the thematic cluster is also therefore tagged without the user having to do this separately?

**Comment by tmfrnz** [comment]
Yes, currently the recommendations have to be assigned separately in addition to assigning the cluster.

It is important to recall that the action-cluster association was introduced at a later stage (while we were with you in Samoa actually) as we saw that in case recommendations were broad and had multiple clusters assigned, specific actions would wrongly be associated (by inheritance) with irrelevant clusters. We had then (in Samoa) also discussed options on how the action-cluster association could be used to facilitate associating actions with recommendations of the same cluster but the challenge identified was that sometimes the action only applies to a subset of the recommendations they share clusters with and we have yet to come to a conclusion on how this can best be achieved.

Very related to this (and somewhat the other way around) is the discussion we had around the double grouping (#9) where we saw challenges when actions implicitly inherit issues from associated recommendations, but again have yet to come to a conclusion.

---

So overall and more generally speaking the challenge is two-fold:
1. how can a user selectively associate entities (eg actions) with other entities (eg recommendations) that share a category (eg cluster)?
2. how can a user selectively associate entities (eg actions) with categories (eg issues) that are associated with associated entities (eg recommendations)?

**Comment by tmfrnz** [comment]
An attempt to describe a possible solution:

### Example: user adds thematic cluster to action:

1. User adds "Thematic cluster" 

If cluster has recommendations that are not previously assigned to action:

2. App prompts user "Would you also like to select related recommendations"?
3. A dialogue opens with a list of recommendations not previously assigned to action
4. User can select or unselect all or specific recommendations

If any of the recommendations have any categories of common taxonomies that are not previously assigned to action

5. App prompts user "Would you also like to select connected categories"?
6. A dialogue opens with a list of connected categories associated with selected recommendations and not previously assigned to action
7. User can select or unselect all or specific categories

If any new category has any recommendations that are not previously assigned to action:

8. Go back to 2.

### Example: user adds recommendation to action:

1. User adds "Recommendation" 

If recommendations has any categories of common taxonomies that are not previously assigned to action

2. App prompts user "Would you also like to select connected categories"?
3. A dialogue opens with a list of connected categories associated with selected recommendations and not previously assigned to action
4. User can select or unselect all or specific categories

If any new category has any recommendations that are not previously assigned to action:

5. App prompts user "Would you also like to select related recommendations"?
6. A dialogue opens with a list of recommendations not previously assigned to action
7. User can select or unselect all or specific recommendations

If any new recommendation has any categories that are not previously assigned to action:

8. Go back to 2.
